code/ui/button/model/button_model.lua

```diff
@@ -48,40 +24,43 @@ function button_model:remove_listener(event_type, callback)
   end
 end
 
-function button_model:try_button_click(x, y, btn, is_touch, is_pressing)
-  if btn == BUTTON_CLICK_TYPES.LEFT and self.rectangle:is_inside(x, y) and is_pressing then
+function button_model:try_button_click(screen_x, screen_y, btn, is_touch, is_pressing)
+  local world_x, world_y = camera:screen_to_world(screen_x, screen_y)
+  if btn == BUTTON_CLICK_TYPES.LEFT and self.rectangle:is_inside(world_x, world_y) and is_pressing then
```

Comment:
om du byter plats på is_inside() checken och is_pressing så behöver det inte räkna ut om det är innuti ifall vi inte clickar ("and" evaluatear inte nästa argument ifall det föregående är false)

Reply:
Snyggt hittat, fixar det!